src/main/java/stats/utils/Point.java

```diff
@@ -0,0 +1,131 @@
+package stats.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+public class Point {
+
+
+    /**
+     *
+     * Creates a List of random Points.
+     */
+    public static List<Point> createRandomPoints(int min, int max, int number) {
+
+        List<Point> points = new ArrayList(number);
+        for (int i = 0; i < number; i++) {
+            points.add(createRandomPoint(min, max));
+        }
+        return points;
+    }
+
+
+    /**
+     * Creates a random Point.
+     * @param min minimum x coordinate.
+     * @param max maximum y coordinate.
+     * @return A random Point.
+     */
+    public static Point createRandomPoint(int min, int max) {
+
+        Random random = new Random();
+        double x = min + (max - min) * random.nextDouble();
+        double y = min + (max - min) * random.nextDouble();
+
+        return new Point(x, y);
+    }
+
+
+    /**
+     * Calculates the Euclidean distance between two Points.
+     *
+     * @param p        The first Point.
+     * @param centroid The second Point.
+     * @return The distance.
+     */
+    public static double distance(Point p, Point centroid) {
```

Comment:
name the function parameters as ```p1``` and ```p2``` ```centroid``` is too application specific to be used here

Reply:
I don't think you overall need this function since its name should be ```euclideanDistance``` and not simply ```distance```